src/main/java/uk/gov/companieshouse/pscstatementdataapi/serialization/LocalDateDeserializer.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.companieshouse.pscstatementdataapi.serialization;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.github.dockerjava.api.exception.BadRequestException;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+
+public class LocalDateDeserializer extends JsonDeserializer {
+    public static final String APPLICATION_NAME_SPACE = "disqualified-officers-data-api";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(APPLICATION_NAME_SPACE);
+
+    @Override
+    public LocalDate deserialize(JsonParser jsonParser, DeserializationContext
+            deserializationContext) {
+        try {
+            DateTimeFormatter dateTimeFormatter = DateTimeFormatter
+                    .ofPattern("yyyy-MM-dd'T'HH:mm:ss'Z'");
+            JsonNode jsonNode = jsonParser.readValueAsTree();
+            JsonNode dateNode = jsonNode.get("$date");
+
+            return dateNode.textValue() != null ?
+                    LocalDate.parse(dateNode.textValue(), dateTimeFormatter) :
+                    LocalDate.ofInstant(Instant.ofEpochMilli(dateNode.get("$numberLong").asLong()), ZoneId.systemDefault());
+        } catch (Exception exception) {
+            LOGGER.error("Deserialization failed.", exception);
+            throw new BadRequestException(exception.getMessage());
```

Comment:
Not a fan of using an external BadRequestException for this. Can we use our own BadRequestException and build it ourselves?